pallets/identity/src/lib.rs

```diff
@@ -403,13 +401,34 @@ decl_module! {
             let sender_key = AccountKey::try_from(ensure_signed(origin)?.encode())?;
             let did = Context::current_identity_or::<Self>(&sender_key)?;
             let _grants_checked = Self::grant_check_only_master_key(&sender_key, did)?;
+            let did_sig = Signatory::from(did);
 
-            // Remove any Pre-Authentication & link
-            signers_to_remove.iter().for_each(|signer| {
-                Self::remove_pre_join_identity(signer, did);
-                if let Signatory::AccountKey(ref key) = signer {
-                    Self::unlink_key_to_did(key, did);
-                }
+            // Remove links and get all authorization IDs per signer.
+            let signer_and_auth_id_list = signers_to_remove.iter().map(|signer| {
+                match signer {
+                    Signatory::AccountKey(ref key) => Self::unlink_key_to_did(key, did),
+                    _ => {}
+                };
+
+                // It returns the list of `auth_id` from `did`.
+                let auth_ids = <Authorizations<T>>::iter_prefix(signer)
+                    .filter_map( |authorization| {
+                        if authorization.authorized_by == did_sig {
```

Comment:
I don't think we need to worry about auths in this function but if we do, we should also filter out auths from master key of the did since most auths are issued by the master key and not did itself.